core/services/chainlink/config.go

```diff
@@ -49,40 +51,208 @@ type Config struct {
+		chainID := config.ChainID()
+		i := slices.IndexFunc(*rs, func(r RawConfig) bool {
+			otherChainID := r.ChainID()
+			return otherChainID != "" && chainID == otherChainID
+		})
+		if i != -1 {
+			if err := (*rs)[i].SetFrom(config); err != nil {
+				return err
+			}
+		} else {
+			*rs = append(*rs, config)
+		}
+	}
+
+	return nil
+}
+
+func (rs RawConfigs) ValidateConfig() (err error) {
+	for _, config := range rs {
+		if configErr := config.ValidateConfig(); configErr != nil {
+			err = errors.Join(err, configErr)
+		}
+	}
+
+	chainIDs := commonconfig.UniqueStrings{}
+	for i, config := range rs {
+		chainID := config.ChainID()
+		if chainIDs.IsDupe(&chainID) {
+			err = errors.Join(err, commonconfig.NewErrDuplicate(fmt.Sprintf("%d.ChainID", i), chainID))
+		}
+	}
+
+	nodeNames := commonconfig.UniqueStrings{}
+	for i, config := range rs {
+		configNodeNames := config.NodeNames()
+		for j, nodeName := range configNodeNames {
+			if nodeNames.IsDupe(&nodeName) {
+				err = errors.Join(err, commonconfig.NewErrDuplicate(fmt.Sprintf("%d.Nodes.%d.Name", i, j), nodeName))
+			}
+		}
+	}
+	return
+}
+
 // RawConfig is the config used for chains that are not embedded.
 type RawConfig map[string]any
 
-// ValidateConfig returns an error if the Config is not valid for use, as-is.
-func (c *RawConfig) ValidateConfig() (err error) {
-	if v, ok := (*c)["Enabled"]; ok {
+type parsedRawConfig struct {
+	chainID     string
+	nodeConfigs []map[string]any
+	nodeNames   []string
+}
+
+func (c RawConfig) parse() (*parsedRawConfig, error) {
+	var err error
+	if v, ok := c["Enabled"]; ok {
 		if _, ok := v.(bool); !ok {
 			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "Enabled", Value: v, Msg: "expected bool"})
 		}
 	}
-	if v, ok := (*c)["ChainID"]; ok {
-		if _, ok := v.(string); !ok {
-			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "ChainID", Value: v, Msg: "expected string"})
+
+	parsedRawConfig := &parsedRawConfig{}
+	chainID, exists := c["ChainID"]
+	if !exists {
+		err = multierr.Append(err, commonconfig.ErrMissing{Name: "ChainID", Msg: "required for all chains"})
+	} else {
+		chainIDStr, ok := chainID.(string)
+		switch {
+		case !ok:
+			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "ChainID", Value: chainID, Msg: "expected string"})
+		case chainIDStr == "":
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "ChainID", Msg: "required for all chains"})
+		default:
+			parsedRawConfig.chainID = chainIDStr
+		}
+	}
+	nodes, exists := c["Nodes"]
+	if !exists {
+		err = multierr.Append(err, commonconfig.ErrMissing{Name: "Nodes", Msg: "expected at least one node"})
+	} else {
+		nodeMaps, ok := nodes.([]any)
+		switch {
+		case !ok:
+			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "Nodes", Value: nodes, Msg: "expected array of node configs"})
+		case len(nodeMaps) == 0:
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "Nodes", Msg: "expected at least one node"})
```

Comment:
I think this is also too strict when called from `SetFrom`. We could claim that NOPs _shoudn't_ do this because it has no effect and could be removed, but it would be cleaner to stay flexible.